src/vma/sock/sockinfo_udp.cpp

```diff
@@ -1581,9 +1582,14 @@ ssize_t sockinfo_udp::tx(const tx_call_t call_type, const struct iovec* p_iov, c
 			si_udp_logdbg("MSG_OOB not supported in UDP (tx-ing to os)");
 			goto tx_packet_to_os;
 		}
-
+		
 		sock_addr dst((struct sockaddr*)__dst);
-
+	
```

Comment:
1. Adding this code (find_target_family()...) here will hurt latency in UDP fast path.
2. It seems that ROLE_UDP_SENDER is already supported in code.
Checking of ROLE_UDP_SENDER already exists in dst_entry_udp.cpp
```code
transport_t dst_entry_udp::get_transport(sockaddr_in to)
{
	return  __vma_match_udp_sender(TRANS_VMA, mce_sys.app_id, (sockaddr *)(&to), sizeof to);
}
```

get_transport() method is a nested call - the result of dst_entry_udp::slow_send()

```code2
ssize_t dst_entry_udp::slow_send(const iovec* p_iov, size_t sz_iov, bool b_blocked /*= true*/, bool is_rexmit /*= false*/, int flags /*= 0*/, socket_fd_api* sock /*= 0*/, tx_call_t call_type /*= 0*/)
{
	NOT_IN_USE(is_rexmit);

	ssize_t ret_val = 0;

	dst_udp_logdbg("In slow send");

	prepare_to_send();  // <========================== m_b_force_os is set to true

	if (m_b_force_os || !m_b_is_offloaded) {
		struct sockaddr_in to_saddr;
		to_saddr.sin_port = m_dst_port;
		to_saddr.sin_addr.s_addr = m_dst_ip.get_in_addr();
		to_saddr.sin_family = AF_INET;
		dst_udp_logdbg("Calling to tx_os");
                //   <=============================== calling OS on next line
		ret_val = sock->tx_os(call_type, p_iov, sz_iov, flags, (const struct sockaddr*)&to_saddr, sizeof(struct sockaddr_in)); 
	}
        ...............
```

Please update libvma.conf file such that by default DNS protocol is not offloaded. Use ROLE_UDP_SENDER